Since 1.2.1: error Could not open cafile: EISDIR: illegal operation on a directory, read

**What is the current behavior?**

Using yarn with workspaces while building Theia I now get erros like:

```
@theia/cli: > yarn run clean && yarn run build
@theia/cli: npm info lifecycle @theia/cli@0.1.1~prepare: @theia/cli@0.1.1
@theia/cli: error Could not open cafile: EISDIR: illegal operation on a directory, read
@theia/cli: error Could not open cafile: EISDIR: illegal operation on a directory, read

```

**If the current behavior is a bug, please provide the steps to reproduce.**

You can reproduce with;

```
git clone https://github.com/theia-ide/theia \
&& cd theia \
&& yarn 
```

**What is the expected behavior?**

With yarn 1.0.2 I do not get this error.

**Please mention your node.js, yarn and operating system version.**
npm --version
5.3.0
OS: linux


Note you can also use this Dockerfile to reproduce:

```
FROM node:8
RUN useradd --create-home theia
WORKDIR /home/theia
RUN rm -rf /opt/yarn && rm -f /usr/local/bin/yarn && rm -f /usr/local/bin/yarnpkg
RUN apt-get update && apt-get install -y npm && npm install -g yarn@1.2.1
USER theia
RUN git clone --depth 1 https://github.com/theia-ide/theia && \
    cd theia && \
    yarn
EXPOSE 3000
WORKDIR /home/theia/theia/examples/browser
CMD yarn run start

```

You will see the error while doing a `docker build -t theia .`

Oops, my fault... I saw 2 bugs with `ESIDIR` errors filed less than an hour apart and assumed they were the same. I'll re-open.  I started trying to reproduce your case above but I get different errors:

```
$ run prepare
lerna info version 2.2.0
@theia/ext-scripts: > @theia/ext-scripts@0.1.1 prepare /Users/jvalore/Projects/theia/dev-packages/ext-scripts
@theia/ext-scripts: > echo 'skip'
@theia/cli: > @theia/cli@0.1.1 prepare /Users/jvalore/Projects/theia/dev-packages/cli
@theia/cli: > yarn run clean && yarn run build
@theia/ext-scripts: skip
@theia/core: > @theia/core@0.1.1 prepare /Users/jvalore/Projects/theia/packages/core
@theia/core: > yarn run clean && yarn run build
@theia/core: [lint]
@theia/core: [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/messaging/proxy-factory.ts[95, 15]: missing whitespace
@theia/core: [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[69, 15]: missing whitespace
@theia/core: [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[90, 15]: missing whitespace
@theia/core: [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[110, 15]: missing whitespace
@theia/core: [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[130, 15]: missing whitespace
@theia/core: [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[150, 15]: missing whitespace
@theia/core: [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[170, 15]: missing whitespace
@theia/core: [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[190, 15]: missing whitespace
@theia/core: [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[198, 15]: missing whitespace
@theia/core: [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[218, 15]: missing whitespace
@theia/core: [lint]
@theia/core: [lint] theiaext lint exited with code 2
@theia/core: [compile] theiaext compile exited with code 0
@theia/core: error Command failed with exit code 1.
@theia/core: npm ERR! code ELIFECYCLE
@theia/core: npm ERR! errno 1
@theia/core: npm ERR! @theia/core@0.1.1 prepare: `yarn run clean && yarn run build`
@theia/core: npm ERR! Exit status 1
@theia/core: npm ERR!
@theia/core: npm ERR! Failed at the @theia/core@0.1.1 prepare script.
@theia/core: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
@theia/core: npm ERR! A complete log of this run can be found in:
@theia/core: npm ERR!     /Users/jvalore/.npm/_logs/2017-10-12T12_05_23_768Z-debug.log
lerna ERR! execute Error: Command failed: npm run prepare
lerna ERR! execute error Command failed with exit code 1.
lerna ERR! execute npm ERR! code ELIFECYCLE
lerna ERR! execute npm ERR! errno 1
lerna ERR! execute npm ERR! @theia/core@0.1.1 prepare: `yarn run clean && yarn run build`
lerna ERR! execute npm ERR! Exit status 1
lerna ERR! execute npm ERR!
lerna ERR! execute npm ERR! Failed at the @theia/core@0.1.1 prepare script.
lerna ERR! execute npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
lerna ERR! execute
lerna ERR! execute npm ERR! A complete log of this run can be found in:
lerna ERR! execute npm ERR!     /Users/jvalore/.npm/_logs/2017-10-12T12_05_23_768Z-debug.log
lerna ERR! execute
lerna ERR! execute
lerna ERR! execute > @theia/core@0.1.1 prepare /Users/jvalore/Projects/theia/packages/core
lerna ERR! execute > yarn run clean && yarn run build
lerna ERR! execute
lerna ERR! execute [lint]
lerna ERR! execute [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/messaging/proxy-factory.ts[95, 15]: missing whitespace
lerna ERR! execute [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[69, 15]: missing whitespace
lerna ERR! execute [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[90, 15]: missing whitespace
lerna ERR! execute [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[110, 15]: missing whitespace
lerna ERR! execute [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[130, 15]: missing whitespace
lerna ERR! execute [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[150, 15]: missing whitespace
lerna ERR! execute [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[170, 15]: missing whitespace
lerna ERR! execute [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[190, 15]: missing whitespace
lerna ERR! execute [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[198, 15]: missing whitespace
lerna ERR! execute [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[218, 15]: missing whitespace
lerna ERR! execute [lint]
lerna ERR! execute [lint] theiaext lint exited with code 2
lerna ERR! execute [compile] theiaext compile exited with code 0
lerna ERR! execute
lerna ERR! execute     at Promise.all.then.arr (/Users/jvalore/Projects/theia/node_modules/execa/index.js:236:11)
lerna ERR! execute     at <anonymous>
lerna ERR! execute     at process._tickCallback (internal/process/next_tick.js:188:7)
lerna ERR! execute  callback with error
lerna ERR! execute  { Error: Command failed: npm run prepare
lerna ERR! execute error Command failed with exit code 1.
lerna ERR! execute npm ERR! code ELIFECYCLE
lerna ERR! execute npm ERR! errno 1
lerna ERR! execute npm ERR! @theia/core@0.1.1 prepare: `yarn run clean && yarn run build`
lerna ERR! execute npm ERR! Exit status 1
lerna ERR! execute npm ERR!
lerna ERR! execute npm ERR! Failed at the @theia/core@0.1.1 prepare script.
lerna ERR! execute npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
lerna ERR! execute
lerna ERR! execute npm ERR! A complete log of this run can be found in:
lerna ERR! execute npm ERR!     /Users/jvalore/.npm/_logs/2017-10-12T12_05_23_768Z-debug.log
lerna ERR! execute
lerna ERR! execute
lerna ERR! execute > @theia/core@0.1.1 prepare /Users/jvalore/Projects/theia/packages/core
lerna ERR! execute > yarn run clean && yarn run build
lerna ERR! execute
lerna ERR! execute [lint]
lerna ERR! execute [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/messaging/proxy-factory.ts[95, 15]: missing whitespace
lerna ERR! execute [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[69, 15]: missing whitespace
lerna ERR! execute [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[90, 15]: missing whitespace
lerna ERR! execute [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[110, 15]: missing whitespace
lerna ERR! execute [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[130, 15]: missing whitespace
lerna ERR! execute [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[150, 15]: missing whitespace
lerna ERR! execute [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[170, 15]: missing whitespace
lerna ERR! execute [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[190, 15]: missing whitespace
lerna ERR! execute [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[198, 15]: missing whitespace
lerna ERR! execute [lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[218, 15]: missing whitespace
lerna ERR! execute [lint]
lerna ERR! execute [lint] theiaext lint exited with code 2
lerna ERR! execute [compile] theiaext compile exited with code 0
lerna ERR! execute
lerna ERR! execute     at Promise.all.then.arr (/Users/jvalore/Projects/theia/node_modules/execa/index.js:236:11)
lerna ERR! execute     at <anonymous>
lerna ERR! execute     at process._tickCallback (internal/process/next_tick.js:188:7)
lerna ERR! execute   stack: 'Error: Command failed: npm run prepare\nerror Command failed with exit code 1.\nnpm ERR! code ELIFECYCLE\nnpm ERR! errno 1\nnpm ERR! @theia/core@0.1.1 prepare: `yarn run clean && yarn run build`\nnpm ERR! Exit status 1\nnpm ERR! \nnpm ERR! Failed at the @theia/core@0.1.1 prepare script.\nnpm ERR! This is probably not a problem with npm. There is likely additional logging output above.\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     /Users/jvalore/.npm/_logs/2017-10-12T12_05_23_768Z-debug.log\n\n\n> @theia/core@0.1.1 prepare /Users/jvalore/Projects/theia/packages/core\n> yarn run clean && yarn run build\n\n[lint] \n[lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/messaging/proxy-factory.ts[95, 15]: missing whitespace\n[lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[69, 15]: missing whitespace\n[lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[90, 15]: missing whitespace\n[lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[110, 15]: missing whitespace\n[lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[130, 15]: missing whitespace\n[lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[150, 15]: missing whitespace\n[lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[170, 15]: missing whitespace\n[lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[190, 15]: missing whitespace\n[lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[198, 15]: missing whitespace\n[lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[218, 15]: missing whitespace\n[lint] \n[lint] theiaext lint exited with code 2\n[compile] theiaext compile exited with code 0\n\n    at Promise.all.then.arr (/Users/jvalore/Projects/theia/node_modules/execa/index.js:236:11)\n    at <anonymous>\n    at process._tickCallback (internal/process/next_tick.js:188:7)',
lerna ERR! execute   code: 1,
lerna ERR! execute   killed: false,
lerna ERR! execute   stdout: '\n> @theia/core@0.1.1 prepare /Users/jvalore/Projects/theia/packages/core\n> yarn run clean && yarn run build\n\n[lint] \n[lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/messaging/proxy-factory.ts[95, 15]: missing whitespace\n[lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[69, 15]: missing whitespace\n[lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[90, 15]: missing whitespace\n[lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[110, 15]: missing whitespace\n[lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[130, 15]: missing whitespace\n[lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[150, 15]: missing whitespace\n[lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[170, 15]: missing whitespace\n[lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[190, 15]: missing whitespace\n[lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[198, 15]: missing whitespace\n[lint] ERROR: /Users/jvalore/Projects/theia/packages/core/src/common/logger.ts[218, 15]: missing whitespace\n[lint] \n[lint] theiaext lint exited with code 2\n[compile] theiaext compile exited with code 0\n',
lerna ERR! execute   stderr: 'error Command failed with exit code 1.\nnpm ERR! code ELIFECYCLE\nnpm ERR! errno 1\nnpm ERR! @theia/core@0.1.1 prepare: `yarn run clean && yarn run build`\nnpm ERR! Exit status 1\nnpm ERR! \nnpm ERR! Failed at the @theia/core@0.1.1 prepare script.\nnpm ERR! This is probably not a problem with npm. There is likely additional logging output above.\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     /Users/jvalore/.npm/_logs/2017-10-12T12_05_23_768Z-debug.log\n',
lerna ERR! execute   failed: true,
lerna ERR! execute   signal: null,
lerna ERR! execute   cmd: 'npm run prepare',
lerna ERR! execute   timedOut: false }
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

I did a `git clone` then a `yarn install` and got the above.

--- a/__tests__/config.js
+++ b/__tests__/config.js
@@ -1,0 +1,44 @@
+/* @flow */
+
+import Config from '../src/config.js';
+import {ConsoleReporter} from '../src/reporters/index.js';
+
+const stream = require('stream');
+
+const initConfig = async cfg => {
+  const stdout = new stream.Writable({
+    decodeStrings: false,
+    write(data, encoding, cb) {
+      cb();
+    },
+  });
+
+  const reporter = new ConsoleReporter({stdout, stderr: stdout});
+  const config = new Config(reporter);
+  await config.init(cfg);
+  return config;
+};
+
+test('getOption changes ~ to cwd when resolve=true', async () => {
+  const config = await initConfig({});
+  config.registries.yarn.config.cafile = '~/';
+  expect(config.getOption('cafile', true)).not.toContain('~');
+});
+
+test('getOption does not change ~ when resolve=false', async () => {
+  const config = await initConfig({});
+  config.registries.yarn.config.cafile = '~/';
+  expect(config.getOption('cafile', false)).toEqual('~/');
+});
+
+test('getOption does not change empty-string when resolve=true', async () => {
+  const config = await initConfig({});
+  config.registries.yarn.config.cafile = '';
+  expect(config.getOption('cafile', true)).toEqual('');
+});
+
+test('getOption does not change empty-string when resolve=false', async () => {
+  const config = await initConfig({});
+  config.registries.yarn.config.cafile = '';
+  expect(config.getOption('cafile', false)).toEqual('');
+});

--- a/src/config.js
+++ b/src/config.js
@@ -195,7 +195,7 @@ export default class Config {
   getOption(key: string, resolve: boolean = false): mixed {
     const value = this.registries.yarn.getOption(key);
 
-    if (resolve && typeof value === 'string') {
+    if (resolve && typeof value === 'string' && value.length) {
       return resolveWithHome(value);
     }
 

